### PR TITLE
Handle INACTIVE role in gateway and clients

### LINKED ISSUE #6053: Handle INACTIVE partition role in gateway
**Describe the bug**
When a partition fails to recover or goes unhealthy the partition switches to the role `INACTIVE`. The gateway cannot handle the `INACTIVE` role and fails to decode it and show it correctly in the topology.

**To Reproduce**

Break a partition or corrupt a snapshot so the partition cannot recover.

**Expected behavior**

The gateway can handle `INACTIVE` partitions.

**Log/Stacktrace**
If possible add the full stacktrace or Zeebe log which contains the issue. 

<details><summary>Full Stacktrace</summary>
{
  "insertId": "nprv5kyjydx40a283",
  "jsonPayload": {
    "serviceContext": {
      "service": "zeebe",
      "version": "0.25.3"
    },
    "context": {
      "threadName": "Broker-0-zb-actors-0",
      "threadId": 28,
      "loggerName": "io.zeebe.protocol",
      "actor-name": "GatewayTopologyManager",
      "threadPriority": 5
    },
    "message": "Failed to decode broker info, found unknown partition role: INACTIVE",
    "logger": "io.zeebe.protocol",
    "thread": "Broker-0-zb-actors-0"
  },
  "timestamp": "2021-01-04T08:46:45.794372Z",
  "severity": "WARNING",
  "logName": "projects/camunda-cloud-240911/logs/stdout",
  "sourceLocation": {
    "file": "BrokerInfo.java",
    "line": "443",
    "function": "lambda$consumePartitions$1"
  },
  "receiveTimestamp": "2021-01-04T08:46:49.049897806Z"
}
<p>

```
<STACKTRACE>
```

</p>
</details>

**Environment:**
 - OS: [e.g. Linux]
 - Zeebe Version: [e.g. 0.20.0]
 - Configuration: [e.g. exporters etc.]

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Handles INACTIVE role in the broker and gateway topology managers, as well as in the clients. Tried it with `zbctl` and it works: 
```
Brokers:
  Broker 0 - mp-spike-zeebe-0.mp-spike-zeebe.mp-spike.svc.cluster.local:26501
    Version: 0.27.0-SNAPSHOT
    Partition 1 : Inactive, Unhealthy
```
## Related issues

closes #6053

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [x] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Review by saig0 — APPROVED** [review]
Looks good :+1: 

I've only one optional suggestion.

File: gateway/src/main/java/io/zeebe/gateway/EndpointManager.java [hunk 1]
```diff
@@ -110,6 +100,33 @@ private void addPartitionInfoToBrokerInfo(
...
+  /**
+   * Sets the broker's partition role in the Partition.Builder
+   *
+   * @return true if it could set the role. False if no role was could be found.
+   */
+  private boolean setRole(
+      final Integer brokerId,
+      final Integer partitionId,
+      final BrokerClusterState topology,
+      final Partition.Builder partitionBuilder) {
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply gateway/src/main/java/io/zeebe/gateway/EndpointManager.java:112]: I tried it but didn't it was much clearer. It becomes longer on the caller's side and you still wonder why an optional builder is being returned (and assigning it into the builder isn't necessary but, if you just call `isEmpty`, then it makes less sense). At least, this is quick to read I think. I think for now it's ok to stay with it but let me know if you strongly disagree

**Comment by MiguelPires** [comment]
Thanks for the quick review @saig0

**Pull request merged**